### DISCUSSION TITLE
Is the `labels` slot must be integer?

### DISCUSSION THREAD

**Discussion by Mr-Milk** [scverse/spatialdata#709] [Q&A]
It's stated in the doc that labels' dtype must be integer, is the data type strictly validated? Will it cause future issues if I store it in float?

**Comment by LucaMarconato** [comment]
Yes, currently it's not validated. I tracked this here: https://github.com/scverse/spatialdata/issues/527.

We will add a validation for integer types because this is expected in several parts of the codebase (so please use integers for labels). Currently validation would pass, but downstream methods such as `aggregate()`, `get_centroids()`, `to_polygons()`, ... would likely fail.
